--- libf2c/f2c.h0	1999-09-07 13:46:24.000000000 -0400
+++ libf2c/f2c.h0	2022-05-01 13:48:12.000000000 -0400
@@ -7,15 +7,15 @@
 #ifndef F2C_INCLUDE
 #define F2C_INCLUDE
 
-typedef long int integer;
-typedef unsigned long int uinteger;
+typedef int integer;
+typedef unsigned int uinteger;
 typedef char *address;
 typedef short int shortint;
 typedef float real;
 typedef double doublereal;
 typedef struct { real r, i; } complex;
 typedef struct { doublereal r, i; } doublecomplex;
-typedef long int logical;
+typedef  int logical;
 typedef short int shortlogical;
 typedef char logical1;
 typedef char integer1;
@@ -42,9 +42,10 @@
 typedef short ftnlen;
 typedef short ftnint;
 #else
-typedef long int flag;
-typedef long int ftnlen;
-typedef long int ftnint;
+typedef int flag;
+typedef int ftnlen;
+typedef int ftnint;
+typedef long ftnlint;
 #endif
 
 /*external read, write*/
